src/sentry/snuba/metrics/extraction.py

```diff
@@ -1042,12 +1081,12 @@ def __init__(
         environment: Optional[str] = None,
         groupbys: Optional[Sequence[str]] = None,
         spec_type: MetricSpecType = MetricSpecType.SIMPLE_QUERY,
-        use_updated_env_logic: bool = True,
+        spec_version: Optional[SpecVersion] = None,
     ):
         self.field = field
         self.query = query
         self.spec_type = spec_type
-        self.use_updated_env_logic = use_updated_env_logic
+        self.spec_version = spec_version if spec_version else SPEC_VERSIONS[1]
```

Comment:
I am not sure I would default to a specific version, seems oddly specific. If you don't pass a spec version, we will default to the minimum version defined by the system. In this case 0. It's just to make the system a bit more predictable.

To do it even better, we could define the MIN_VERSION as a global constant that you update whenever you want to start generating specs in another way.